test/test-handlebars.js

```diff
@@ -37,6 +37,25 @@ test('fastify.view with handlebars engine', t => {
   })
 })
 
+test('fastify.view with handlebars engine catches render error', t => {
+  t.plan(2)
+  const fastify = Fastify()
+  const handlebars = require('handlebars')
+
+  handlebars.registerHelper('error', () => { throw new Error('kaboom') })
+
+  fastify.register(require('../index'), {
+    engine: {
+      handlebars: handlebars
+    }
+  })
+
+  fastify.ready(err => {
+    t.error(err)
+    t.rejects(fastify.view('./templates/error.hbs'), /kaboom/)
```

Comment:
I would change this test using `inject` in order to check the response we get from an endpoint